cpp/arcticdb/storage/s3/mock_s3_client.cpp

```diff
@@ -0,0 +1,166 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include <arcticdb/storage/s3/mock_s3_client.hpp>
+#include <arcticdb/storage/s3/s3_client_wrapper.hpp>
+
+#include <arcticdb/util/preconditions.hpp>
+#include <arcticdb/util/pb_util.hpp>
+#include <arcticdb/log/log.hpp>
+#include <arcticdb/util/buffer_pool.hpp>
+
+#include <arcticdb/storage/storage_utils.hpp>
+
+#include <aws/s3/S3Errors.h>
+
+namespace arcticdb::storage{
+
+using namespace object_store_utils;
+
+namespace s3 {
+
+std::string operation_to_string(S3Operation operation){
+    switch (operation) {
+        case S3Operation::HEAD: return "Head";
+        case S3Operation::GET: return "Get";
+        case S3Operation::PUT: return "Put";
+        case S3Operation::DELETE: return "Delete";
+        case S3Operation::DELETE_LOCAL: return "Delete_local";
+        case S3Operation::LIST: return "List";
+    }
+    util::raise_rte("Invalid s3 operation");
+}
+
+std::optional<S3Operation> string_to_operation(const std::string& str){
```

Comment:
What's this one for? I would think we could always pass in the enum rather than strings?